src/test/java/org/datadog/jmxfetch/TestReconnectContainer.java

```diff
@@ -0,0 +1,200 @@
+import java.net.URL;
+import java.nio.file.Paths;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.datadog.jmxfetch.reporter.ConsoleReporter;
+
+@Slf4j
+class JMXFailServerClient {
+    private final String host;
+    private final int port;
+
+    public JMXFailServerClient(String host, int port) {
+        this.host = host;
+        this.port = port;
+    }
+
+    public void startJMXServer() throws IOException {
+        sendPostRequest("/start");
+    }
+
+    public void stopJMXServer() throws IOException {
+        sendPostRequest("/stop");
+    }
+
+    public void jmxCutNetwork() throws IOException {
+        sendPostRequest("/jmx/cutNetwork");
+    }
+
+    public void jmxRestoreNetwork() throws IOException {
+        sendPostRequest("/jmx/restoreNetwork");
+    }
+
+    private void sendPostRequest(String endpoint) throws IOException {
+        URL url = new URL("http://" + host + ":" + port + endpoint);
+        log.info("Sending POST to {} ", url);
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setRequestMethod("POST");
+
+        int responseCode = con.getResponseCode();
+
+        if (responseCode == HttpURLConnection.HTTP_OK) {
+            BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+            String inputLine;
+            StringBuffer response = new StringBuffer();
+            while ((inputLine = in.readLine()) != null) {
+                response.append(inputLine);
+            }
+            in.close();
+            System.out.println(response.toString());
+        } else {
+            System.out.println("HTTP POST request failed with status code: " + responseCode);
+        }
+    }
+}
+
+
+public class TestReconnectContainer extends TestRemoteAppCommon {
+    private static final int rmiPort = 9090;
+    private static final int controlPort = 9091;
+
+    private static boolean isDomainPresent(String domain, MBeanServerConnection mbs) {
+        boolean found = false;
+        try {
+            String[] domains = mbs.getDomains();
+            for (int i = 0; i < domains.length; i++) {
+                if (domains[i].equals(domain)) {
+                    found = true;
+                }
+            }
+        } catch (IOException e) {
+            found = false;
+        }
+        return found;
+    }
+
+    private static ImageFromDockerfile img = new ImageFromDockerfile()
+        .withFileFromPath(".", Paths.get("./tools/misbehaving-jmx-server/"));
+
+    @Rule
+    public GenericContainer<?> cont = new GenericContainer<>(img)
+        .withExposedPorts(rmiPort, controlPort)
+        .withEnv(Collections.singletonMap("JJ_RMI_PORT", "" + rmiPort))
+        .withEnv(Collections.singletonMap("JJ_CONTROL_HTTP_PORT", "" + controlPort))
+        .waitingFor(Wait.forLogMessage(".*IAMREADY.*", 1));
+
+    @Test
+    public void testJMXDirectBasic() throws Exception {
```

Comment:
These "jmx direct" tests were useful for my development, but they may not be useful to hold on to for the final PR

Reply:
I'll leave them, they don't hurt anything